include/libpmemfile-posix.h

```diff
@@ -158,9 +157,47 @@ extern "C" {
 typedef struct pmemfilepool PMEMfilepool;
 typedef struct pmemfile_file PMEMfile;
 
+typedef unsigned pmemfile_mode_t;
+typedef unsigned pmemfile_uid_t;
+typedef unsigned pmemfile_gid_t;
+typedef long long pmemfile_ssize_t;
+typedef long long pmemfile_off_t;
+typedef unsigned long pmemfile_nlink_t;
+typedef long pmemfile_blksize_t;
+typedef long pmemfile_blkcnt_t;
+typedef unsigned long pmemfile_dev_t;
+typedef unsigned long pmemfile_ino_t;
+
+struct pmemfile_timespec
+{
+	long tv_sec;
+	long tv_nsec;
+};
+
+struct pmemfile_stat
+{
+	pmemfile_dev_t st_dev;
+	pmemfile_ino_t st_ino;
+	pmemfile_nlink_t st_nlink;
```

Comment:
If we change the type of this field to unsigned, then pmemfile_stat size and layout won't match native Linux struct stat (and pmemfile_inode already has nlink of type uint64_t, so...).
Assuming you meant unsigned long long (which I just realized pmemfile_nlink_t should be instead of current "unsigned long"), I think we should have pmemfile versions of all types if we want applications using pmemfile-posix to be portable.